base_exception/models/base_exception.py

```diff
@@ -63,6 +72,19 @@ class ExceptionRule(models.Model):
 #  - context: current context
 """)
 
+    @api.multi
+    def _get_domain(self):
+        """ override me to customize domains according exceptions cases """
+        self.ensure_one()
+        return safe_eval(self.domain)
+
+    @api.onchange('exception_type',)
+    def onchange_exception_type(self):
+        if self.exception_type == 'by_domain':
```

Comment:
You shouldn't empty fields, or you will lose possible store things. You have to check the type to know which field to use, but not clear it IMO.